packages/util-endpoints/src/utils/evaluateTemplate.spec.ts

```diff
@@ -0,0 +1,52 @@
+import { getAttr } from "../lib";
+import { evaluateTemplate } from "./evaluateTemplate";
+
+jest.mock("../lib");
+
+describe(evaluateTemplate.name, () => {
+  const mockOptions = {
+    endpointParams: {},
+    referenceRecord: {},
+  };
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe("should replace `{parameterName}` with value", () => {
+    const parameterName = "bar";
+    const template = "foo {parameterName} baz";
+
+    afterEach(() => {
+      expect(getAttr).not.toHaveBeenCalled();
+    });
+
+    it.each(["endpointParams", "referenceRecord"])("from %s", (key: string) => {
+      expect(evaluateTemplate(template, { ...mockOptions, [key]: { parameterName } })).toBe(`foo ${parameterName} baz`);
+    });
+  });
+
+  it("should not replace string escaped {{value}}", () => {
+    const value = "bar";
+    expect(evaluateTemplate("foo {{value1}} bar {{value2}} baz", { ...mockOptions, endpointParams: { value } })).toBe(
+      "foo {value1} bar {value2} baz"
+    );
```

Comment:
Am I too nit-picky here? Does it need to handle any even numbers of bracket? Should the template `{{{{value}}}}` replaced to `"{{value}}"`(escape every 2 brackets) or `"{{{value}}}"`(only escape inner-most 2 brackets)?

Reply:
Not sure. I've created an internal discussion for this edge case to update the specification. I'll follow-up if needed.